### PR TITLE
Make toolchain resolution message clearer.

### LINKED ISSUE #26319: Bad error message when using the wrong Java version
### Current Behavior

simdjson-java ( https://github.com/simdjson/simdjson-java ) is the Java port of the industry-leading json library.

simdjson-java requires Java 20, as specified:

```
java {
    toolchain {
        languageVersion = JavaLanguageVersion.of(20)
    }
    withJavadocJar()
    withSourcesJar()
}
```

However, when running tests with an older Java, the error message is over-complicated and opaque, especially for those less familiar with Java versioning.

F5XKZOGWUAEQwM5

### Expected Behavior

Java version mis-match is a common problem with any JVM build tool. Gradle should emit a simple and concise message like "Found Java X, required Java Y".

### Context (optional)

_No response_

### Steps to Reproduce

Run the tests (or any build step) in https://github.com/simdjson/simdjson-java/tree/main

### Gradle version

8.2

### Build scan URL (optional)

_No response_

### Your Environment (optional)

_No response_

### PR DESCRIPTION
**Pull request by reinsch82**
Fix #26319

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by reinsch82** [comment]
@bot-gradle test this

**Review by donat — APPROVED** [review]

**Comment by reinsch82** [comment]
@bot-gradle test and merge

**Comment by reinsch82** [comment]
@bot-gradle test this

**Comment by reinsch82** [comment]
@bot-gradle test and merge

**Review by octylFractal — APPROVED** [review]

**Comment by donat** [comment]
@reinsch82 the failure is valid, the expected output of the sample test should be adjusted: https://github.com/gradle/gradle/blob/6b3ce84c34812851fe144f99c0efda27ff28bdab/subprojects/docs/src/snippets/java/toolchain-filters/tests/matchingVendorWithoutCC.out#L5-L4

**Comment by reinsch82** [comment]
Thx working in it 🙂

**Comment by reinsch82** [comment]
@bot-gradle merge

File: subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/logging/PlatformInOutputNormalizer.groovy [hunk 1]
```diff
@@ -15,20 +15,23 @@
...
-class ToolchainDownloadOutputNormalizer implements OutputNormalizer {
+class PlatformInOutputNormalizer implements OutputNormalizer {
+    def static systemInfo = NativeServices.createSystemInfo()
+    def static os = DefaultBuildPlatform.getOperatingSystem(OperatingSystem.current())
+
     @Override
     String normalize(String commandOutput, ExecutionMetadata executionMetadata) {
         return commandOutput
-            .replaceAll(
-                "No matching toolchains found for requested specification: (.+) for (.+).") {
```
Comment (donat) [inline subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/logging/PlatformInOutputNormalizer.groovy:30]: Why was it so specific? 😠

**Comment by reinsch82** [comment]
@bot-gradle cancel

**Comment by reinsch82** [comment]
@bot-gradle merge

**Comment by reinsch82** [comment]
@bot-gradle test this

**Comment by reinsch82** [comment]
@bot-gradle test this

**Comment by reinsch82** [comment]
@bot-gradle merge

**Pull request merged**